### ISSUE TITLE
loss nan

### ISSUE THREAD

**Issue by Jianf-l** [NamGyuKang/PIXEL#3]
Dear authors:
I really appreciate your innovative idea, and thanks for opening source code. I am currently following your work, but when i run the code on Burgers with L-BFGS optimizer, i find that PIXEL can reach 3e-2 L2 loss after about 2000 epochs, however the loss (both f-loss and data-loss) suddenly becomes nan at this point, which doesn't recover and thus cann't reach 9.98e-4 reported in the paper (n-cells=16). What are the possible reasons?

**Comment by NamGyuKang** [comment]
Hello. Thank you for reaching out.

Did you use the command provided on the GitHub page?
If you run the following command:

"python main.py --network base --pde burgers_1d --in-dim 2 --out-dim 1 --use-cell --n-cells 16 --cell-dim 4 --cell-size 16 --interp cosine --num-train 100000 --num-init 100000 --random-f --f-scale 0.01 --hidden-dim 16 --num-layers 2 --max-iter 39001 --seed 500 --tag sample_tag"

With this command (seed = 500), PIXEL achieved an L2 error of about 9.33e-04 at 315 epoches.

-> Iter 316, Loss: 1.48060e-04, Loss_u: 7.82052e-09, Loss_b: 0.00000e+00, Loss_f: 1.48052e-04, Loss_tv: 0.00000e+00
-> [Test Iter:315, 12_Loss: 9.32645e-04]

-> Iter 526, Loss: 1.23892e-04, Loss_u: 7.62948e-09, Loss_b: 0.00000e+00, Loss_f: 1.23884e-04, Loss_tv: 0.00000e+00
-> [Test Iter:525, 12_Loss: 6.42780e-04]

--> Iter 2011, Loss: 9.70933e-05, Loss_u: 3.93306e-09, Loss_b: 0.00000e+00, Loss_f: 9.70894e-05, Loss_tv: 0.00000e+00                                                                 
--> [Test Iter:2010, 12_Loss: 7.20514e-04] 

Regarding the NaN issue, this can occur if--random-f command is not enabled.
The --random-f option resamples the collocation points every epoch.

May I ask which seed you used? In addition, regarding your comment about data loss, are you currently incorporating a data loss term in your training setup for an inverse problem?

If you have any further questions, please feel free to reach out. I’d be happy to help.

Sincerely,
Namgyu Kang

**Comment by Jianf-l** [comment]
Thank you for your response!
Actually, I used the commands provided by GitHub Pages, yet replacing ‘--num points 100000 --init points 100000’ with ‘--num points 10000 --init points 10000’ to expedite obtaining my experimental results.
I know this might increase the loss, but may not  be the reason the loss became NaN. I may have made formatting or function modifications to your code for my own convenience, inadvertently altering the original logic. I will re-clone your repository and conduct experiments following the instructions in the README.
Thanks for your help again!

**Comment by NamGyuKang** [comment]
Thank you very much for your interest in my research.

Based on my experience, PIXEL does not train well with only 10,000 collocation points.

For 16x16 cells, since the architecture divides the domain into 16×16 cells, each cell requires a sufficient number of collocation points to achieve good learning performance. Therefore, providing around 100,000 points is highly recommended to ensure improve training accuracy.

If you would like to obtain faster experimental results, I would recommend considering the use of CosineSampler cuda code for interpolation. CosineSampler can reduce both memory consumption and computational time.

If you encounter any issues during installation or have any further questions, please feel free to let me know.

Thank you again for taking the time to run experiments with my work.

Sincerely,
Namgyu Kang
